tools/asset-sync/assets.ps1

```diff
@@ -0,0 +1,521 @@
+Set-StrictMode -Version 3
+
+# See https://stackoverflow.com/a/14440066 for a succinct explanation. We will be using
+# this methodology for any global variables that can be considered constants
+. (Join-Path $PSScriptRoot ".." ".." "eng" "common" "scripts" "common.ps1")
+
+$REPO_ROOT = Resolve-Path (Join-Path $PSScriptRoot ".." "..")
+$ASSETS_STORE = (Join-Path $REPO_ROOT ".assets")
+
+<#
+.SYNOPSIS
+Checks the contents of a directory, then returns a tuple of booleans @("assetsJsonPresent", "isRootFolder").
+
+.DESCRIPTION
+Evaluates a directory by checking its contents. First value of the tuple is whether or not a "assets.json" file 
+is present A "root" directory is one where a assets.json is present OR where we are as far up the file tree as 
+we can possibly ascend. 
+
+.PARAMETER TargetPath
+The path we will evaulate to detect either a root directory or a .git folder.
+#>
+Function Evaluate-Target-Dir {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $TargetPath
+    )
+
+    $isFile = Test-Path $TargetPath -PathType Leaf
+    $foundConfig = $false
+    $foundRoot = $false
+
+    if ($isFile) {
+        throw "Evaluated a file `"$TargetPath`" as a directory. Exiting."
+    }
+
+    # we need to force to ensure we grab hidden files in the directory dump
+    $files = Get-ChildItem -Force $TargetPath
+
+    foreach($file in $files)
+    {
+        if($file.Name.ToLower() -eq "assets.json"){
+            $foundConfig = $true
+        }
+
+        # need to ensure that if we're outside a git repo, we will still break out eventually.
+        if(($file.Name.ToLower() -eq ".git") -or -not (Split-Path $TargetPath)){
```

Comment:
What case is `Split-Path` supposed to detect here?

Reply:
Also I wonder if it's simpler to specifically check `Get-ChildItem` for `$TargetPath/assets.json` and `$TargetPath/.git`?

Reply:
The latter case is ensuring we're dealing with a directory, not a single file.

`Split-Path "world.json" -> ""`

Reply:
I am going to jump on your suggestion for specific GCI though.